src/api/v4/dataview/base.js

```diff
@@ -214,19 +215,19 @@ Base.prototype._initialize = function (source, column, options) {
 
 Base.prototype._checkSource = function (source) {
   if (!(source instanceof SourceBase)) {
-    throw new TypeError('Source property is required.');
+    throw this._getValidationError('sourceRequired');
```

Comment:
Having a key + message for each and every possible "validation" error seems like a lot of work to me. You might want to kill me, but I personally think that a TypeError + some string interpolation might just be okay in this case where a user just needs a very basic message error. Something like rails does (`validate_presence_of :source`) would be nice.

Reply:
Basically, we could build something a bit more generic that didn't force us to define (and maintain) a key and a friendly message.

Reply:
Is not so hard to maintain, instead throwing an error with a single string every time, just create a pair key, message.
Doing this we get lot of advantages, we could give the user an url with info to solve the error, we get better error classification, less chances of different logs for the same error..